lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GUESecurityCapability.cpp

```diff
@@ -45,6 +45,7 @@ int UESecurityCapabilityMsg::DecodeUESecurityCapabilityMsg(
   MLOG(MDEBUG) << " length = " << hex << int(ue_sec_capability->length) << endl;
 
   // 5GS encryption algorithms
+  ea                     = *(buffer + decoded);
   ue_sec_capability->ea0 = (*(buffer + decoded) >> 7) & 0x1;
   ue_sec_capability->ea1 = (*(buffer + decoded) >> 6) & 0x1;
   ue_sec_capability->ea2 = (*(buffer + decoded) >> 5) & 0x1;
```

Comment:
compiler may optimize this, but could be more straightforward to write:
ue_sec_capability->ea0 = (ea >> 7) & 0x1;

Reply:
Fixed